WonderMake/Utilities/Container/Container.h

```diff
@@ -0,0 +1,144 @@
+#pragma once
```

Comment:
We can probably constexpr a few more functions. But this is dependant on the Storage type, which might not support it. Any idea on how to fix that?

Reply:
Will separate container backends to different classes in the future which might make it easier but unsure. Too much of the design is subject to change atm.